addons/web_tour/static/tests/tour_service.test.js

```diff
@@ -931,3 +931,54 @@ test("manual tour with inactive steps", async () => {
     expect(".counter .value").toHaveText("5");
     await advanceTime(10000);
 });
+
+test("automatic tour with alternative trigger", async () => {
+    expect.assertions(6);
+    patchWithCleanup(browser.console, {
+        log: (s) => {
+            expect(s.includes("on step") || s.toLowerCase().includes("succeeded")).toBe(true);
+        },
+        warn: (s) => expect(s.length).toBe(true),
+        error: (s) => expect(s.length).toBe(true),
```

Comment:
how can this be true? length is a number, not true

Reply:
console.warn and console.error are not expected. So these expects are normally not called and if they are, the test fails.

Reply:
throw error instead then

Reply:
okay ... :)